src/main/java/org/reaktivity/nukleus/kafka/internal/stream/NetworkConnectionPool.java

```diff
@@ -2230,26 +2161,24 @@ private NetworkTopic(
             this.partitions = new TreeSet<>();
             this.candidate = new NetworkTopicPartition();
             this.progressHandler = this::handleProgress;
-            PartitionIndex[] partitionIndexes = new PartitionIndex[partitionCount];
 
             if (compacted)
             {
-                for (int i = 0; i < partitionCount; i++)
-                {
-                    partitionIndexes[i] = new CompactedPartitionIndex(1000, deleteRetentionMs,
-                            messageCache);
-                }
+                cache = new CompactedTopicCache(
+                                partitionCount,
+                                deleteRetentionMs,
+                                messageCache,
+                                clientStreamFactory.counters.cacheHits,
+                                clientStreamFactory.counters.cacheMisses);
             }
             else
             {
-                for (int i = 0; i < partitionCount; i++)
-                {
-                    partitionIndexes[i] = DEFAULT_PARTITION_INDEX;
-                }
+                cache = DefaultTopicCache.INSTANCE;
```

Comment:
Rename to `EmptyTopicCache` or `StreamingTopicCache`.

Reply:
done